### DISCUSSION TITLE
curl syntax for search

### DISCUSSION THREAD

**Discussion by rebred** [pocketbase/pocketbase#5895] [Q&A] [answered]
I am trying to search a record where the `username` is `Jack`
`

curl -X GET http://127.0.0.1:8089/api/collections/db/records \
   -H "Content-Type: application/json" \
   -d '{"filter": "(username='Jack')"}'

`
what would be the proper syntax ?

**Comment by chnfyi** [comment, marked as answer]
You can try the cURL below. Filter should be an url query string instead of a body request.

_I assumed that you serve effectively PocketBase on port 8089 and that `Jack` belongs to `users` collection so you'll need to replace it with your actual collection if it's different._

```
curl GET 'http://127.0.0.1:8089/api/collections/users/records?filter=(username="Jack")' -H "Content-Type: application/json"
```

**Reply by rebred** [reply]
thanks a lot it worked!!

if I may ask how do I add

`
username="Jack"&&created>"2022-01-01"

`

**Reply by chnfyi** [reply]
Yes it should work as you wrote it.

```
curl GET 'http://127.0.0.1:8089/api/collections/users/records?filter=(username="Jack"&&created>"2022-01-01")' -H "Content-Type: application/json"
```

**Reply by rebred** [reply]
`
curl GET 'http://127.0.0.1:8089/api/collections/db/records?filter=(name="Jack" && created>"2022-01-01")' -H "Content-Type: application/json"
`

I get:

curl: (6) Could not resolve host: GET
curl: (3) URL using bad/illegal format or missing URL


-----

`
curl GET 'http://127.0.0.1:8089/api/collections/db/records?filter=(name="Jack"&&created>"2022-01-01")' -H "Content-Type: application/json"
`

I get:

curl: (6) Could not resolve host: GET
{"code":400,"message":"Something went wrong while processing your request.","data":{}}

**Reply by ganigeorgiev** [reply]
The URL query parameters must be URL encoded (the JS and Dart SDKs do this automatically).

The correct URL in your case should be:
```
http://127.0.0.1:8089/api/collections/db/records?filter=name%3D%22Jack%22%20%26%26%20created%3E%222022-01-01%2023%3A59%3A59.999Z%22
``` 
(_side-note: dates are compared as strings and the full PocketBase date format is: `Y-m-d H:i:s.uZ`, e.g. `2022-01-01 23:59:59.999Z`_)

**Reply by rebred** [reply]
CURL has the command:

`--data-urlencode`

would you know where to plug it in the CURL command ?

**Reply by ganigeorgiev** [reply]
~`--data-urlencode` is for encoding a body content for sending `application/x-www-form-urlencoded` or similar is not for the query parameters (or at least I don't see how it could be used this way).~ _Edit: See below._

The requirement for encoded URL query parameters is not PocketBase-only specific.

If you need further help, please elaborate what actually you are trying to do.

**Reply by ganigeorgiev** [reply]
Actually when testing locally, it seems that `--data-urlencode` indeed can be used for encoding the query params, aka. try with:

```
curl --get \
    --data-urlencode "filter=(name='Jack' && created>'2022-01-01 23:59:59.999Z')" \
    http://127.0.0.1:8089/api/collections/db/records
```

**Reply by rebred** [reply]
great thanks a lot!!
